### PR TITLE
refactor: improvements after removing simplebar dep

### PR DESCRIPTION
**Pull request by vanbasten17**
This is a continuation of: https://github.com/hubtype/botonic/pull/2900
Several things done here:
* Remove `scrollable-content` generic component and create it for each scrollable div.
* Add notable refs (`webchatRef`, `chatAreaRef`, `headerRef`) in `WebchatContext` to be accessed from different parts of the coee
* Remove JS commented code already migrated to TS.
* Reduce `util/dom.js` in favour of using new hooks.
* Migrate from JS to TS:
   * `device-adapter`
   * `scrollbar-controller`
   * `webchat-resizer`
* Simplify `WebchatHeader` component
* New hooks:
   * `useScrollToBottom`: centralize logic for scrolling to the end of scrollable messages list container
   * `useVirtualKeyboardDetection`: detect whether virtual keyboard is open or not (iOS/Safari)
   * `useWebchatDimensions`: recalculation of webchat / chat area heights 
* Create a new component `WebchatChatArea`
* Set textarea fontsize to 16px to be consistent with the rest of styles and avoid auto-focusing issues on iOS.

### CODE REVIEW

File: packages/botonic-react/src/webchat/webchat-chat-area/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React, { useContext, useEffect, useState } from 'react'
+
+import { WebchatContext } from '../../contexts'
+import { BotonicContainerId } from '../constants'
+import { useWebchatDimensions } from '../hooks'
+import { WebchatMessageList } from '../message-list'
+import { WebchatReplies } from '../replies'
+import { StyledWebchatChatArea } from './styles'
+
+export const WebchatChatArea = () => {
```
Comment (Iru89) [inline packages/botonic-react/src/webchat/webchat-chat-area/index.tsx:10]: for a future refactor rename this component to ChatArea and WebchatInputPanel to InputPanel

File: packages/botonic-react/src/webchat/hooks/use-scroll-to-bottom.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export const useScrollToBottom = ({
+  host,
+  behavior = 'smooth',
+  timeout = 200,
+}) => {
+  const scrollToBottom = () => {
+    const webchatElement = getWebchatElement(host)
+    if (!webchatElement) return
+    const scrollableMessagesList = document.getElementById(
```
Comment (Iru89) [inline packages/botonic-react/src/webchat/hooks/use-scroll-to-bottom.ts:12]: investigate why you need to work with id instead of ref in this hook

**Review by Iru89 — APPROVED** [review]

File: packages/botonic-react/src/webchat/hooks/use-scrollbar-controller.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
+import { useContext, useEffect } from 'react'
+
+import { WebchatContext } from '../../contexts'
+import { DEVICES, isMobileDevice } from '../devices'
+
+// TODO: Investigate why when we have some messages, scroll actions are not disabled properly
+
+const debounced = (delay, fn) => {
```
[the remark this answers is not in the document]
Comment (vanbasten17) [reply packages/botonic-react/src/webchat/hooks/use-scrollbar-controller.ts:8]: investigate in another PR

File: packages/botonic-react/src/webchat/webchat-chat-area/index.tsx (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-react/src/webchat/webchat-chat-area/index.tsx:10]: next PR

File: packages/botonic-react/src/webchat/hooks/use-scroll-to-bottom.ts (hunk 2 above)
Comment (vanbasten17) [reply packages/botonic-react/src/webchat/hooks/use-scroll-to-bottom.ts]: fixed in last commit

**Pull request merged**